### PR TITLE
[SOC-5562]Improvement in user suggestions when @mentioning in the Activity Stream

### PR DESCRIPTION
**Pull request by AtiaZ**
fix description : 

1- add a new typeofRelation "mention_activity_stream" for the mentioning in the activity stream.
2- Illustrate the connections firstly and then the others users.

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java [hunk 1]
```diff
@@ -339,6 +341,22 @@ public Response suggestUsernames(@Context UriInfo uriInfo,
...
 
+    } else if (MENTION_ACTIVITY_STREAM.equals(typeOfRelation)) {
+      ListAccess<Identity> connections = getRelationshipManager().getConnectionsByFilter(currentIdentity, identityFilter);
+      int size = connections.getSize();
+      List<UserInfo> userInfos = new ArrayList<PeopleRestService.UserInfo>();
+      if (connections != null && connections.getSize() > 0) {
+        Identity[] identities = connections.load(0, size < SUGGEST_LIMIT ? size : (int) SUGGEST_LIMIT);
+        userInfos = addUsersToUserInfosList(identities, excludedIdentityList, userInfos);
+      }
+      ListAccess<Identity> listAccess = getIdentityManager().getIdentitiesByProfileFilter(OrganizationIdentityProvider.NAME, identityFilter, false);
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: We should check if the suggestion limit is not already reached before retrieving all the users IMO.

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java [hunk 2]
```diff
@@ -339,6 +341,22 @@ public Response suggestUsernames(@Context UriInfo uriInfo,
         }
       }
 
+    } else if (MENTION_ACTIVITY_STREAM.equals(typeOfRelation)) {
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: Please add comments (in which case are you, the big steps).

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java [hunk 3]
```diff
@@ -347,8 +365,25 @@ public Response suggestUsernames(@Context UriInfo uriInfo,
     return Util.getResponse(nameList, uriInfo, mediaType, Response.Status.OK);
   }
 
-  private void addSpaceOrUserToList(List<Identity> identities, IdentityNameList options,
-                                   Space space, String typeOfRelation, int order) throws SpaceException {
+  private List<UserInfo> addUsersToUserInfosList(Identity[] identities, List<Identity> excludedIdentityList, List<UserInfo> userInfos) {
+    UserInfo userInfo;
+    String userType = ConversationState.getCurrent().getIdentity().getUserId();
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: Should be renamed to currentUserId IMO.

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java (hunk 2 above)
Comment (AtiaZ) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: done

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java (hunk 3 above)
Comment (AtiaZ) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: done.

**Comment by thomasdelhomenie** [comment]
The PR looks good but it needs to be updated after merge of https://jira.exoplatform.org/browse/SOC-5563.

**Pull request closed without merging**